src/libraries/System.Text.Json/docs/ReferenceHandling_spec.md

```diff
@@ -0,0 +1,810 @@
+}
+```
+
+# Other languages
+
+## Newtonsoft.Json
+Newtonsoft.Json contains settings that you can enable to deal with such problems.
+* For Serialization:
+  * [`ReferenceLoopHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_ReferenceLoopHandling.htm)
+  * [`PreserveReferencesHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_PreserveReferencesHandling.htm)
+* For Deserialization:
+  * [`MetadataPropertyHandling`](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_MetadataPropertyHandling.htm)
+
+When using `ReferenceLoopHandling.Ignore`, other objects that were already seen on the current graph branch will be ignored on serialization.
+
+When using `PreserveReferencesHandling.All` you are signaling that your resulting JSON will contain *metadata* properties `$ref`, `$id` and `$values` which are going to act as reference identifiers (`$id`) and pointers (`$ref`). 
+Now, to read back those references, you have to use `MetadataPropertyHandling.Default` to indicate that *metadata* is expected in the payload passed to the `Deserialize` method.
+
+* Pros
+  * If we opt-in for this we could provide compatibility with Newtonsoft which is always desired by the community.
+* Cons
+  * Quite invasive, (it affects `JsonException.Path`, `JsonSerializerOptions.IgnoreNullValues`, `JsonPropertyNameAttribute`, and Converters).
+  * This would break existing converters i.e: an array converter may expect the first token to be "[" and a preserved array starts with "{".
+    * perhaps converters are more feasible with the JSON path implementation.
+  * We will now accept that an array comes in valid format when starts with a curly brace "{"; below issue is related to guard against NRE when this happens:
+    * https://github.com/dotnet/corefx/issues/41839
+
+## dojo toolkit (JavaScript framework)
+https://dojotoolkit.org/reference-guide/1.10/dojox/json/ref.html
+
+Similar: https://www.npmjs.com/package/json-cyclic 
+
+* id-based (ignore this approach since is the same the one of Newtonsoft.Json)
+* path-based
+  * "\#" denotes the root of the object and then uses semantics inspired in JSONPath.
+  * It does not uses `$id` nor `$values` metadata, therefore, everything can be referenced.
+  * Pros
+    * It looks cleaner. 
+    * Only disruptive (weird) edge case would be a reference to an array i.e: { "MyArray": { "$ref": "#manager.subordinates" } }.
+  * Cons
+    * Path value will become too long on very deep objects.
+    * Storing all the complex types could become very expensive, are we going to store also primitive types?
+    * This would break existing converters when handling reference to an array.
+    * Not compatible with Newtonsoft.Json.
+  
+## flatted (JavaScript module) (probably not worth it)
+https://github.com/WebReflection/flatted
+
+* While stringifying, all Objects, including Arrays, and strings, are flattened out and replaced as unique index.
```

Comment:
```suggestion
* While stringifying, all Objects, including Arrays and strings, are flattened out and replaced with unique index.
```

Reply:
I extracted that paragraph from the official GitHub repo, but ok, I can take the suggestion.
https://github.com/WebReflection/flatted#how-does-it-work-